src/Core/Data/TripleBuffer.hpp

```
@@ -0,0 +1,49 @@
+    bool consume() {
+        // "Acquire" ensures that once we see the new index, we also see the data that was written to that buffer.
+        int newest = m_dirtyIndex.exchange(m_readIndex, std::memory_order_acquire);
+        if (newest == m_readIndex)
+            // Nothing new was published
+            return false;
+
+        m_readIndex = newest;
+        return true;
+    }
+
+    T &writeRef() { return m_buffers[m_writeIndex]; }
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)